sqlx-core/src/any/row.rs

```diff
@@ -120,8 +120,8 @@ impl AnyRow {
                 _ if value.is_null() => AnyValueKind::Null,
                 AnyTypeInfoKind::Null => AnyValueKind::Null,
                 AnyTypeInfoKind::Bool => AnyValueKind::Bool(decode(value)?),
-                AnyTypeInfoKind::SmallInt => AnyValueKind::SmallInt(decode(value)?),
-                AnyTypeInfoKind::Integer => AnyValueKind::Integer(decode(value)?),
+                AnyTypeInfoKind::SmallInt => AnyValueKind::BigInt(decode(value)?),
+                AnyTypeInfoKind::Integer => AnyValueKind::BigInt(decode(value)?),
                 AnyTypeInfoKind::BigInt => AnyValueKind::BigInt(decode(value)?),
                 AnyTypeInfoKind::Real => AnyValueKind::Real(decode(value)?),
                 AnyTypeInfoKind::Double => AnyValueKind::Double(decode(value)?),
```

Comment:
My best guess is that the same happens for floating point numbers, but I'm not entirely sure how I'd test loss of precision.